consensus/state.go

```diff
@@ -1033,10 +1077,11 @@ func (cs *ConsensusState) AddProposalBlockPart(height int, part *types.Part) (ad
 		log.Info("Received complete proposal", "hash", cs.ProposalBlock.Hash())
 		if cs.Step == RoundStepPropose && cs.isProposalComplete() {
 			// Move onto the next step
+			// XXX: isn't this unecessary since propose will either do this or timeout into it
```

Comment:
EnterPropose only begins the timeout && begins disseminating the proposed block, but early termination of the Propose step only happens asynchronously upon receiving the final piece of data, which is either a block part or a POL vote.  So this is necessary, unless we have some ticker (which is not implemented anywhere) which repeatedly checks for IsProposalComplete() but I think this is better.